usaspending_api/etl/elasticsearch_loader_helpers/controller.py

```diff
@@ -93,31 +96,34 @@ def complete_process(self) -> None:
             )
             update_last_load_date(f"{self.config['stored_date_key']}", self.config["processing_start_datetime"])
 
-    def construct_tasks(self):
+    def determine_partitions(self) -> int:
+        """Create partion size less than or equal to max_size for more even distribution"""
+        if self.config["partition_size"] > self.record_count:
+            return 1
+        return ceil(self.record_count / self.config["partition_size"])
 
+    def construct_tasks(self) -> List[TaskSpec]:
+        """Create the Task objects w/ the appropriate configuration"""
         name_gen = gen_random_name()
         return [self.configure_task(j, name_gen) for j in range(self.config["partitions"])]
 
-    def configure_task(self, number: int, name_gen) -> TaskSpec:
-        sql_str = EXTRACT_SQL.format(
-            divisor=self.config["partitions"],
-            id_col=self.config["primary_key"],
-            remainder=number,
-            update_date=self.config["starting_date"],
-            view=self.config["sql_view"],
-        )
+    def configure_task(self, number: int, name_gen: Generator) -> TaskSpec:
```

Comment:
In lieu of docstring, it would help me and others looking at configuring a task if this were named `partition_number` instead of just `number`, to give some sense that this task is around a single partition of data.

Reply:
fair point